### PR TITLE
Update documentation to reflect the {state: "all"} option

### PR DESCRIPTION
**Pull request by nilbus**
This applies to both issues and milestones where the default state is "open".
Although undocumented, the "all" option is confirmed to work.

See the API docs:
  * https://developer.github.com/v3/issues/#parameters-1
  * https://developer.github.com/v3/issues/milestones/#parameters

### CODE REVIEW

File: lib/octokit/client/milestones.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def list_milestones(repository, options = {})
       # @param repository [String, Repository, Hash] A GitHub repository.
       # @param options [Hash] A customizable set of options.
       # @option options [Integer] :milestone Milestone number.
-      # @option options [String] :state (open) State: <tt>open</tt> or <tt>closed</tt>.
+      # @option options [String] :state (open) State: <tt>open</tt>, <tt>closed</tt>, or <tt>all</tt>.
```
Comment (pengwynn) [inline lib/octokit/client/milestones.rb]: Oops, everything but `number` is existing copypasta. Mind removing `:state`, `:sort`, and `:direction here?

File: lib/octokit/client/milestones.rb [hunk 2]
```diff
@@ -44,7 +44,7 @@ def milestone(repository, number, options = {})
       # @param repository [String, Repository, Hash] A GitHub repository.
       # @param title [String] A unique title.
       # @param options [Hash] A customizable set of options.
-      # @option options [String] :state (open) State: <tt>open</tt> or <tt>closed</tt>.
+      # @option options [String] :state (open) State: <tt>open</tt>, <tt>closed</tt>, or <tt>all</tt>.
```
Comment (pengwynn) [inline lib/octokit/client/milestones.rb]: The `all` value works for listings, but isn't a valid value for specifying state on a milestone.

File: lib/octokit/client/milestones.rb [hunk 3]
```diff
@@ -61,7 +61,7 @@ def create_milestone(repository, title, options = {})
       # @param number [String, Integer] ID of the milestone
       # @param options [Hash] A customizable set of options.
       # @option options [String] :title A unique title.
-      # @option options [String] :state (open) State: <tt>open</tt> or <tt>closed</tt>.
+      # @option options [String] :state (open) State: <tt>open</tt>, <tt>closed</tt>, or <tt>all</tt>.
```
Comment (pengwynn) [inline lib/octokit/client/milestones.rb]: Same here, the `all` value works for listings, but isn't a valid value for specifying state on a milestone.

**Comment by pengwynn** [comment]
Thanks @nilbus. I left a couple of comments.

**Comment by nilbus** [comment]
:dizzy_face: good catch! Thank you for looking so closely. I will update this soon.

**Comment by nilbus** [comment]
Updated! This looks better to me now.

**Comment by pengwynn** [comment]
:cake:

**Pull request merged**